src/main/java/com/zendesk/maxwell/util/ListWithDiskBuffer.java

```diff
@@ -19,33 +19,26 @@
 	private ObjectInputStream is;
 	private ObjectOutputStream os;
 
-	public ListWithDiskBuffer(long maxInMemoryElements) throws IOException {
+	public ListWithDiskBuffer(long maxInMemoryElements) {
 		this.maxInMemoryElements = maxInMemoryElements;
 		list = new LinkedList<>();
 	}
 
 	public void add(T element) throws IOException {
 		list.add(element);
 
-		while (this.list.size() > maxInMemoryElements) {
-			if ( file == null ) {
-				file = File.createTempFile("maxwell", "events");
-				file.deleteOnExit();
-				os = new ObjectOutputStream(new BufferedOutputStream(new FileOutputStream(file)));
-			}
-
-			if ( elementsInFile == 0 )
-				LOGGER.debug("Overflowed in-memory buffer, spilling over into " + file);
-
-			os.writeObject(this.list.removeFirst());
-
-			elementsInFile++;
-
-			if ( elementsInFile % maxInMemoryElements == 0 )
-				os.reset(); // flush ObjectOutputStream caches
-		}
+		while ( shouldBuffer() )
+			evict();
 	}
-
+//*
```

Comment:
sorry, I meant to comment.  the comment markers are not multi-line, though they sorta look like it.  Those comment markers are there to clean up the diff.  For some reason the generic diff algorithm was matching up blank lines that shouldn't have related to each other, making the diff ugly.  The comments markers prevented them from matching, giving a nice continuous diff.  I'll delete them before merge